### PR TITLE
[3802] Setup state & client-test modules and update UT imports

### PR DESCRIPTION
**Pull request by kanat**
Related PR: https://github.com/GetStream/stream-chat-android/pull/3670

### 🎯 Goal

- create `stream-chat-android-state` module
- create `stream-chat-android-client-test` module
- set Gradle dependencies up
- move common test classes from `stream-chat-android-offline` to `stream-chat-android-client-test` 

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

**Review by leandroBorgesFerreira — APPROVED** [review]
Looks ok to me. Another pair of eyes would be also good

File: stream-chat-android-client-test/detekt-baseline.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<SmellBaseline>
+  <ManuallySuppressedIssues/>
+  <CurrentIssues>
+    <ID>MaxLineLength:TestDataHelper.kt$TestDataHelper$MemberRemovedEvent(EventType.MEMBER_REMOVED, Date(), member2.user, channel1.cid, channel1.type, channel1.id, member1)</ID>
+    <ID>MaxLineLength:TestDataHelper.kt$TestDataHelper$checkNotNull(dotenv["STREAM_USER_1_TOKEN"]) { "Be sure to specify the STREAM_USER_1_TOKEN environment variable" }</ID>
+    <ID>MaxLineLength:TestDataHelper.kt$TestDataHelper$public val apiKey: String = checkNotNull(dotenv["STREAM_API_KEY"]) { "Be sure to specify the STREAM_API_KEY environment variable" }</ID>
+    <ID>MaxLineLength:TestDataHelper.kt$TestDataHelper$public val reaction1: Reaction = Reaction("message-1", "like", 1).apply { user = user1; userId = user1.id; score = 10 }</ID>
```
Comment (JcMinarro) [inline stream-chat-android-client-test/detekt-baseline.xml:8]: Can we start the module without Detekt issues?

**Review by JcMinarro — CHANGES_REQUESTED** [review]

**Comment by JcMinarro** [comment]
Ideally we should move to "[Test-Fixtures](https://docs.gradle.org/current/userguide/java_testing.html#sec:java_test_fixtures)", but seems it is not fully supported by [Android Gradle Plugin](https://issuetracker.google.com/issues/139438142) nor [Kotlin Gradle Plugin](https://youtrack.jetbrains.com/issue/KTIJ-14843/Add-support-for-new-Gradle-Test-Fixtures#focus=Comments-27-6062058.0-0) :crying_cat_face:

**Review by JcMinarro — APPROVED** [review]
LGTM

**Pull request merged**